pyaedt/filtersolutions_core/optimization_goals_table.py

```diff
@@ -0,0 +1,411 @@
+        self._dll_interface = pyaedt.filtersolutions_core._dll_interface()
+        self._define_multiple_bands_dll_functions()
+
+    def _define_multiple_bands_dll_functions(self):
+        """Define C++ API DLL functions."""
+        self._dll.getOptimizationGoalDefinitionRowCount.argtype = POINTER(c_int)
+        self._dll.getOptimizationGoalDefinitionRowCount.restype = c_int
+
+        self._dll.getOptimizationGoalDefinitionRow.argtype = [c_int, POINTER(c_char_p), c_int]
+        self._dll.getOptimizationGoalDefinitionRow.restype = c_int
+
+        self._dll.updateOptimizationGoalDefinitionRow.argtype = [c_int, c_int, c_char_p]
+        self._dll.updateOptimizationGoalDefinitionRow.restype = c_int
+
+        self._dll.appendOptimizationGoalDefinitionRow.argtype = [c_int, c_char_p]
+        self._dll.appendOptimizationGoalDefinitionRow.restype = c_int
+
+        self._dll.insertOptimizationGoalDefinitionRow.argtypes = [c_int, c_char_p, c_char_p]
+        self._dll.insertOptimizationGoalDefinitionRow.restype = c_int
+
+        self._dll.removeOptimizationGoalDefinitionRow.argtype = c_int
+        self._dll.removeOptimizationGoalDefinitionRow.restype = c_int
+
+        self._dll.adjustGoalFrequency.argtype = c_char_p
+        self._dll.adjustGoalFrequency.restype = c_int
+
+    @property
+    def row_count(self) -> int:
+        """Number of golas in the optimization goals table.
+        The default is `0`.
+
+        Returns
+        -------
+        int
+        """
+        table_row_count = c_int()
+        status = self._dll.getOptimizationGoalDefinitionRowCount(byref(table_row_count))
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+        return int(table_row_count.value)
+
+    def row(self, row_index) -> list:
+        """Get the row parameters of the optimization goals table as a list.
+
+        Parameters
+        ----------
+        row_index: int
+            Index of the row. Valid values range from ``0`` to ``49``, inclusive.
+
+        Returns
+        -------
+        list
+            A list of strings representing the row parameters.
+        """
+        row_parameter_buffer = create_string_buffer(1024)
+        # Call the DLL function. Assuming it fills the buffer with comma-separated values.
+        status = self._dll.getOptimizationGoalDefinitionRow(row_index, byref(row_parameter_buffer), 1024)
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+        # Decode the buffer to a Python string and split by comma to get a list.
+        row_parameters = row_parameter_buffer.value.decode("utf-8").split("|")
+        return row_parameters
+
+    def update_row(
+        self,
+        row_index,
+        lower_frequency=None,
+        upper_frequency=None,
+        goal_value=None,
+        condition=None,
+        parameter_name=None,
+        weight=None,
+        enabled=None,
+    ):
+        """Update the row parameters for a row in the optimization goals table.
```

Comment:
Is it ok to update a row that's empty? Is it ok to skip rows (update row 2 when there is no entry in row 0 or 1)?